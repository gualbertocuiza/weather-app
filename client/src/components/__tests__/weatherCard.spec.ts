import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import WeatherCard from '../weatherCard.vue'

describe('weather card component', () => {
  const { location, current } = weather
  const wrapper = mount(WeatherCard, {
    props: {
      weather: weather,
    },
  })

  it('should display the city, country', () => {
    expect(wrapper.find('h1').text()).toContain(
      `${location.name}, ${location.country}`
    )
  })

  it('should display the temperature in degrees Celsius by default', () => {
    expect(wrapper.find('#temp').text()).toContain(`${current.temp_c}°C`)
  })

  it('should switch between celcious and fahrenheit with toggle input', async () => {
    const temp = wrapper.find('#temp')
    const feelsLike = wrapper.find('#feels-like')

    expect(temp.text()).toContain(`${current.temp_c}°C`)
    expect(feelsLike.text()).toContain(`${current.feelslike_c}°C`)

    await wrapper.find('input').setValue(true)

    expect(temp.text()).toContain(`${current.temp_f}°F`)
    expect(feelsLike.text()).toContain(`${current.feelslike_f}°F`)
  })
})

const weather = {
  location: {
    name: 'Cochabamba',
    region: 'Cochabamba',
    country: 'Bolivia',
    lat: -17.38,
    lon: -66.15,
    tz_id: 'America/La_Paz',
    localtime_epoch: 1689777381,
    localtime: '2023-07-19 10:36',
  },
  current: {
    last_updated_epoch: 1689777000,
    last_updated: '2023-07-19 10:30',
    temp_c: 17,
    temp_f: 62.6,
    is_day: 1,
    condition: {
      text: 'Sunny',
      icon: '//cdn.weatherapi.com/weather/64x64/day/113.png',
      code: 1000,
    },
    wind_mph: 3.8,
    wind_kph: 6.1,
    wind_degree: 250,
    wind_dir: 'WSW',
    pressure_mb: 1030,
    pressure_in: 30.42,
    precip_mm: 0,
    precip_in: 0,
    humidity: 34,
    cloud: 0,
    feelslike_c: 17,
    feelslike_f: 62.6,
    vis_km: 10,
    vis_miles: 6,
    uv: 4,
    gust_mph: 0.4,
    gust_kph: 0.7,
  },
}
