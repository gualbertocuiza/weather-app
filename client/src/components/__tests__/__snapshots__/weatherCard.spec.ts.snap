// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`weather card component > should render correctly 1`] = `
<div
  class="w-96"
  data-v-88aee66d=""
>
  <div
    class="card w-auto shadow-xl"
    data-v-88aee66d=""
  >
    <div
      class="card-body"
      data-v-88aee66d=""
    >
      <h1
        class="text-3xl"
        data-v-88aee66d=""
      >
        Cochabamba, Bolivia
      </h1>
      <small
        data-v-88aee66d=""
      >
        2023-07-19 10:36 - Sunny
      </small>
      <div
        class="weahter flex justify-between"
        data-v-88aee66d=""
      >
        <h1
          data-testid="temp"
          data-v-88aee66d=""
        >
          17째C
        </h1>
        <img
          alt="Sunny"
          data-v-88aee66d=""
          height="90"
          src="//cdn.weatherapi.com/weather/64x64/day/113.png"
          width="90"
        />
      </div>
      <div
        class="divider"
        data-v-88aee66d=""
      />
      <div
        class="text-sm tracking-wide text-gray-500"
        data-v-88aee66d=""
      >
        <div
          class="flex justify-between mb-2"
          data-v-88aee66d=""
        >
          <span
            data-v-88aee66d=""
          >
            Windy
          </span>
          <span
            data-v-88aee66d=""
          >
            6.1 km/h
          </span>
        </div>
        <div
          class="flex justify-between mb-2"
          data-v-88aee66d=""
        >
          <span
            data-v-88aee66d=""
          >
            Windy Dir
          </span>
          <span
            data-v-88aee66d=""
          >
            WSW
          </span>
        </div>
        <div
          class="flex justify-between mb-2"
          data-v-88aee66d=""
        >
          <span
            data-v-88aee66d=""
          >
            Humidity
          </span>
          <span
            data-v-88aee66d=""
          >
            34
          </span>
        </div>
        <div
          class="flex justify-between mb-2"
          data-v-88aee66d=""
        >
          <span
            data-v-88aee66d=""
          >
            cloud
          </span>
          <span
            data-v-88aee66d=""
          >
            0
          </span>
        </div>
        <div
          class="flex justify-between mb-2"
          data-v-88aee66d=""
        >
          <span
            data-v-88aee66d=""
          >
            Feels like
          </span>
          <span
            data-testid="feels-like"
            data-v-88aee66d=""
          >
            17째C
          </span>
        </div>
      </div>
      <div
        class="divider"
        data-v-88aee66d=""
      />
      <div
        class="card-actions justify-between"
        data-v-88aee66d=""
      >
        <div
          class="flex align-center"
          data-v-88aee66d=""
        >
          <span
            class="text-gray-500"
            data-v-88aee66d=""
          >
            째C
          </span>
          <input
            class="toggle mx-1"
            data-v-88aee66d=""
            type="checkbox"
          />
          <span
            class="text-gray-500"
            data-v-88aee66d=""
          >
            째F
          </span>
        </div>
        <button
          class="btn btn-neutral btn-sm"
          data-v-88aee66d=""
        >
          Full report
        </button>
      </div>
    </div>
  </div>
</div>
`;
